addons/html_builder/static/src/builder/plugins/move/move_plugin.js

```diff
@@ -0,0 +1,192 @@
+import { Plugin } from "@html_editor/plugin";
+import {
+    addMobileOrders,
+    removeMobileOrders,
+} from "@html_builder/builder/utils/column_layout_utils";
+import { isMobileView } from "@html_builder/builder/utils/utils";
+
+// TODO find where to define the selectors so they are not duplicated across files
+const moveUpOrDown = {
+    selector: [
+        "section",
+        ".s_accordion .accordion-item",
+        ".s_showcase .row .row:not(.s_col_no_resize) > div",
+        ".s_hr",
+        // In snippets files
+        ".s_pricelist_boxed_item",
+        ".s_pricelist_cafe_item",
+        ".s_product_catalog_dish",
+        ".s_timeline_list_row",
+        ".s_timeline_row",
+    ].join(", "),
+};
+
+const moveLeftOrRight = {
+    selector: [
+        ".row:not(.s_col_no_resize) > div",
+        ".nav-item",
+        ".s_timeline_card", // timeline TODO custom function, other plugin ?
+    ].join(", "),
+    exclude: ".s_showcase .row .row > div",
+};
+
+export class MovePlugin extends Plugin {
+    static id = "move";
+    static dependencies = ["overlay", "history"];
+    resources = {
+        has_overlay_buttons: this.isTargetMovable.bind(this),
+        get_overlay_buttons: this.getActiveOverlayButtons.bind(this),
+    };
+
+    setup() {
+        this.target = null;
+        this.state = {
+            isMobileView: false,
+            isGridItem: true,
+        };
```

Comment:
i m not a big fan to use "state" when it s not a reactive. It could be confusing